dynamicPermissions.ps1

```diff
@@ -0,0 +1,464 @@
+                        @{
+                            op    = "remove"
+                            path  = "members"
+                            value = @(
+                                @{
+                                    value   = $currentAccount.id
+                                    display = $currentAccount.userName
+                                }
+                            )
+                        }
+                    )
+                }
+
+                $body = ($permissionBody | ConvertTo-Json -Depth 10)
+                $splatWebRequest = @{
+                    Uri             = "$baseUrl/scim/groups/$($permission.Name)"
+                    Headers         = $headers
+                    Method          = 'PATCH'
+                    Body            = ([System.Text.Encoding]::UTF8.GetBytes($body))
+                    ContentType     = "application/json;charset=utf-8"
+                    UseBasicParsing = $true
+                }
+
+                if (-Not($actionContext.DryRun -eq $true)) {
+                    Write-Verbose "Revoking permission: [$($permission.Value) ($($permission.Name))] to account: [$($currentAccount.userName) ($($currentAccount.id))]"
+
+                    $revokePermission = Invoke-RestMethod @splatWebRequest -Verbose:$false
+
+                    $outputContext.AuditLogs.Add([PSCustomObject]@{
+                            # Action  = "" # Optional
+                            Message = "Successfully revoked permission: [$($permission.Value) ($($permission.Name))] to account: [$($currentAccount.userName) ($($currentAccount.id))]"
+                            IsError = $false
+                        })
+                }
+                else {
+                    Write-Warning "DryRun: Would revoke permission: [$($permission.Value) ($($permission.Name))] to account: [$($currentAccount.userName) ($($currentAccount.id))]"
+                }
+            }
+            catch {
+                $ex = $PSItem
+                $errorMessage = Get-ErrorMessage -ErrorObject $ex
+
+                Write-Verbose "Error at Line [$($ex.InvocationInfo.ScriptLineNumber)]: $($ex.InvocationInfo.Line). Error: $($errorMessage.VerboseErrorMessage)"
+                Write-Verbose "URI: $($splatWebRequest.Uri)"
+                Write-Verbose "Body: $($body)"
+
+                $outputContext.AuditLogs.Add([PSCustomObject]@{
+                        # Action  = "" # Optional
+                        Message = "Error revoking permission: [$($permission.Value) ($($permission.Name))] to account: [$($currentAccount.userName) ($($currentAccount.id))]. Error Message: $($errorMessage.AuditErrorMessage)"
+                        IsError = $true
+                    })
+
+                # Throw terminal error
+                throw
+            }
+        }
+        else {
+            $newCurrentPermissions[$permission.Name] = $permission.Value
+        }
+    }
+}
+catch {
+    $ex = $PSItem
```

Comment:
do we need to log this to the auditlogs?

Reply:
Not needed, as the catch is only for the dryrun